syntax = "proto3";
import "google/protobuf/empty.proto";

package FedVectorDB;

service FedSqlService {
    rpc ExecuteKnnQuery(KnnSql) returns (google.protobuf.Empty) {}

    rpc GetKnnCandidate(CandSql) returns (stream VectorDataCand) {}

    rpc ClearKnnCandidate(google.protobuf.Empty) returns (google.protobuf.Empty) {}
};

message KnnSql {
    int64 vid = 1;
    int64 k = 2;
    repeated float data = 3;
};

message CandSql {
    int64 k = 1;
};

message VectorDataCand {
    int64 vid = 1;
    repeated float data = 2;
};

